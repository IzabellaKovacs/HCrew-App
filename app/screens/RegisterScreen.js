import React from 'react';
import { StyleSheet, View, Image, Text, TouchableOpacity, TextInput } from 'react-native';

function LoginScreen() {
    return (
        <View style={styles.container}>
            <Image style={styles.logo} source={require('../assets/lgrg.png')} />
            <TextInput style={{ top: -30, height: 50, width: 300, justifyContent: 'center', backgroundColor: 'white', borderRadius: 15 }}
                placeholder=" Username"
                placeholderTextColor="#771D98"
            />
            <TextInput style={{ top: -20, height: 50, width: 300, justifyContent: 'center', backgroundColor: 'white', borderRadius: 15 }}
                placeholder=" Emaill"
                placeholderTextColor="#771D98"
            />
            <TextInput style={{ top: -10, height: 50, width: 300, justifyContent: 'center', backgroundColor: 'white', borderRadius: 15 }}
                placeholder=" Password"
                placeholderTextColor="#771D98"
            />
            <TouchableOpacity style={[styles.buttonContainer, { top: 30, backgroundColor: 'white', borderWidth: 5, borderColor: '#771D98' }]}>
                <Text style={{ textAlign: 'center', fontSize: 28, top: 0, color: '#771D98', fontWeight: 'bold' }}>Register</Text>
            </TouchableOpacity>

            <Text style={{ top: 40, color: "#771D98", fontWeight: 'bold', fontSize: 20 }}>or with</Text>

            <View style={{ flexDirection: 'row' }} >
                <Image style={{ width: 50, height: 50, position: 'center', top: 48, marginHorizontal: 10 }} source={require('../assets/fb.png')} />
                <Image style={{ width: 50, height: 50, position: 'center', top: 48 }} source={require('../assets/gm.png')} />
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#E7CDEA',
        alignItems: 'center',
        justifyContent: 'center',
    },
    logo: {
        width: 200,
        height: 200,
        position: 'absolute',
        top: 60,
    },
    buttonContainer: {
        top: 30,
        height: 50,
        width: 300,
        justifyContent: 'center',
        backgroundColor: '#771D98',
        borderRadius: 25,
    }
});

export default LoginScreen;
