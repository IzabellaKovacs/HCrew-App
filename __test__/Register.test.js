import React from 'react';
import RegisterScreen from '../app/screens/RegisterScreen';

import { create } from 'react-test-renderer';
import renderer from 'react-test-renderer';

const tree = create(<RegisterScreen />);

test('snapshot renders correctly', () => {
    expect(tree).toMatchSnapshot();
});

it('register: empty email', () => {
    let data = renderer.create(<RegisterScreen />).getInstance();
    window.alert = jest.fn();
    data.signUpUser('','');
    expect(window.alert).toHaveBeenCalledTimes(1);
    expect(window.alert).toHaveBeenCalledWith('Write your email!');
})

it('register: empty password', () => {
    let data = renderer.create(<RegisterScreen />).getInstance();
    window.alert = jest.fn();
    data.setState({email: 'username@yahoo.com'});
    data.signUpUser(data.state.email, '');
    
    expect(window.alert).toHaveBeenCalledTimes(1);
    expect(window.alert).toHaveBeenCalledWith('Write your password!');
})

it('register: success', () => {
    let data = renderer.create(<RegisterScreen />).getInstance();
    data.setState({email: 'username@yahoo.com'});
    data.setState({password: '123456'});
    data.signUpUser(data.state.email, data.state.password);
    
    expect(data.state.signup).toEqual('success');
})
