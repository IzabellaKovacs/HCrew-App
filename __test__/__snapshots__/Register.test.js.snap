// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#E7CDEA",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 200,
        "position": "absolute",
        "top": 60,
        "width": 200,
      }
    }
  />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    onChangeText={[Function]}
    placeholder=" Email"
    placeholderTextColor="#771D98"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 15,
        "height": 50,
        "justifyContent": "center",
        "top": -20,
        "width": 300,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder=" Password"
    placeholderTextColor="#771D98"
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 15,
        "height": 50,
        "justifyContent": "center",
        "top": -10,
        "width": 300,
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "#771D98",
        "borderRadius": 25,
        "borderWidth": 5,
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
        "top": 30,
        "width": 300,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#771D98",
          "fontSize": 28,
          "fontWeight": "bold",
          "textAlign": "center",
          "top": 0,
        }
      }
    >
      Register
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "#771D98",
        "fontSize": 20,
        "fontWeight": "bold",
        "top": 40,
      }
    }
  >
    or with
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 50,
          "marginHorizontal": 10,
          "position": "center",
          "top": 48,
          "width": 50,
        }
      }
    />
    <Image
      source={1}
      style={
        Object {
          "height": 50,
          "position": "center",
          "top": 48,
          "width": 50,
        }
      }
    />
  </View>
</RCTSafeAreaView>
`;
